vignettes/how-gargle-gets-tokens.Rmd

```diff
@@ -81,6 +81,8 @@ For R users, a service account is a great option for credentials that will be us
 
 The JSON key file must be managed securely. In particular, it should not be kept in, e.g., a GitHub repository (unless it is encrypted). The encryption strategy used by gargle and other packages is described in the article [Managing tokens securely](https://gargle.r-lib.org/articles/articles/managing-tokens-securely.html).
 
+Note that fetching a token for a service account requires a reasonably accurate system clock. This is of particular importance for users running gargle inside a Docker container, as Docker for Windows has [intermittently seen problems with clock drift](https://github.com/docker/for-win/issues/4526). If your service account token requets fail with "Bad Request" inside a container but succeed locally, check that the container's system clock is accurate.
```

Comment:
Even more specifically: Docker for Windows on the WSL2 engine.

Reply:
Nah, this affects both the Hyper-V and WSL2 engines. I originally encountered it with Hyper-V; I'm told it also happens on WSL2, although I haven't had the opportunity to test it (I'm on WSL2 now, but I use a work laptop from home now, so it tends to be shut down each day 😛).

Reply:
Okay then wow lol.

For me it's absolutely wild. Clocks out of sync right after system boot sometimes.

Reply:
Yeah, it was pretty wild to diagnose! 😬

Are there any other changes you'd like me to make here?

Reply:
None that I can think of

Reply:
Thanks for taking a look! 😁